src/pages/MapPage.tsx

```diff
@@ -23,7 +43,13 @@ const MapPage: React.FC = () => {
       </IonHeader>
 
       <IonContent fullscreen>
+        <MapButton />
         <MainMap />
+        {window.innerWidth > 768 ? (
+          <BaseLayerPopover />
+        ) : (
+          <BaseLayerSheetModal modal={modal}></BaseLayerSheetModal>
+        )}
```

Comment:
This does not work. On a Linux machine this is always using the Popover. 
Secondly, the render will not be updated when the innerWidth changes. You need to use useEffect, useCallback or a custom hook for that. 
I suggest that you merge, but dirctly open a bug-issue and assign to me. We need an application-wide `useViewport` hook that gives you the width and triggers if the width changes. We need to also integrate Ionic platform information into it, to better distinguish between screen-size and device information and adjust the UI accordingly.

Reply:
It's true that it only works after reloading. But this should be the base case anyway. Normally, the screen size should not change. 
Does this not work on Linux machines?
An application wide hook seems helpful also for further styling.
So, I will merge and open an issue.